File: apps/codemap-search/experiments/event-navigation-poc/language-corpus/fixtures/go/callbacks.go
package callbacks

type Router struct { primary func(); secondary func() }
func (r *Router) setPrimary(cb func()) { r.primary = cb } // store_primary
func (r *Router) setSecondary(cb func()) { r.secondary = cb } // store_secondary
func (r *Router) firePrimary() { r.primary() } // call_primary
func (r *Router) fireSecondary() { r.secondary() } // call_secondary
type Other struct { primary func() }
func (r *Other) firePrimary() { r.primary() } // call_other

File: apps/codemap-search/experiments/event-navigation-poc/language-corpus/fixtures/rust/callbacks.rs
struct Router { primary: fn(), secondary: fn() }
impl Router {
    fn set_primary(&mut self, cb: fn()) { self.primary = cb; } // store_primary
    fn set_secondary(&mut self, cb: fn()) { self.secondary = cb; } // store_secondary
    fn fire_primary(&self) { (self.primary)(); } // call_primary
    fn fire_secondary(&self) { (self.secondary)(); } // call_secondary
}
struct Other { primary: fn() }
impl Other {
    fn fire_primary(&self) { (self.primary)(); } // call_other
}
